components/mpas-framework/src/framework/mpas_field_routines.F

```diff
@@ -2858,6 +2869,183 @@ subroutine mpas_allocate_mold_1dchar(dst, src)!{{{
+            return
+         endif
+      enddo
+
+      call mpas_log_write('Assign pointer from '//trim(field % fieldName)//' to '//trim(mask % fieldName))
+      field % missingValueMask2d => mask
+
+   end subroutine mpas_mask_assign_pointer_field2d_mask2d_real
+
+!***********************************************************************
+!
+!  routine mpas_mask_field3d_mask3d_real
+!
+!> \brief   MPAS 3D real field mask routine.
+!> \author  Carolyn Begeman, Adrian Turner
+!> \date    05/17/22
+!> \details 
+!> Masks the source field using a masking array.
+!
+!-----------------------------------------------------------------------
+!   subroutine mpas_mask_assign_pointer_field3d_mask3d_real(field, mask) !{{{
+!
+!      implicit none
+!
+!      type (field3DReal), intent(inout), pointer :: field
+!      type (field3DInteger), intent(in), pointer :: mask
+!
+!      integer :: i
+!      integer, dimension(3) :: fieldDims
+!      integer, dimension(3) :: maskDims
+!
+!      character (len=StrKIND), dimension(3) :: fieldDimNames
+!      character (len=StrKIND), dimension(3) :: maskDimNames
+!
+!      !if ( field % missingValue == null() ) then
+!      !   call mpas_log_write('Warning: no missingValue was provided for fieldName') ! TODO Replace field name iwth field name
+!      !   return
+!      !endif
+!      maskDims = mask % dimSizes
+!      maskDimNames = mask % dimNames
+!      ! TODO check that dimNames and dimSizes match
+!      do i = 1, 3
+!         if ( maskDimNames(i) .ne. fieldDimNames(i) ) then
+!            call mpas_log_write('Warning: dimension names do not match between masking and output arrays')
+!         endif
+!      enddo
+!
+!      field % missingValueMask3d => mask
+!
+!   end subroutine mpas_mask_assign_pointer_field3d_mask3d_real
+
+!***********************************************************************
+!
+!  routine mpas_mask_field2d_mask2d_real
+!
+!> \brief   MPAS 3D real field mask routine.
+!> \author  Carolyn Begeman, Adrian Turner
+!> \date    05/17/22
+!> \details 
+!> Masks the source field using a masking array.
+!
+!-----------------------------------------------------------------------
+   subroutine mpas_mask_field2d_mask2d_real(field, mask, maskedField) !{{{
+
+      implicit none
+
+      type (field2DReal), intent(in), pointer :: field !< Input/Output: Field to be masked
+
+      type (field2DInteger), intent(in), pointer :: mask
+
+      real (kind=RKIND), dimension(:,:), intent(out) :: maskedField !< Input/Output: Field to be masked
+      ! TODO consider whether this should be of field type or 3d real array
+      !type (field3DReal), intent(out) :: maskedField !< Input/Output: Field to be masked
+
+      real (kind=RKIND), dimension(:,:), pointer :: fieldArray
+      character (len=100) :: log_string
+      integer, dimension(:,:), pointer :: maskArray
+
+      call mpas_log_write('Masking array '//trim(field % fieldName))
+      fieldArray = field % array
```

Comment:
fieldArray is a pointer that has neither been allocated or assigned to another pointer at this point.
This line may need to be fieldArray => field % array to assign the pointer if that was your intent

Reply:
Great, thanks!